src/Sylius/Behat/Context/Api/Shop/ProductContext.php

```diff
@@ -111,6 +111,14 @@ public function iBrowseProductsFromTaxon(?TaxonInterface $taxon = null): void
         }
     }
 
+    /**
+     * @When /^I sort products by the (oldest|newest) date first$/
+     */
+    public function iSortProductsByTheDateFirst(string $sortDirection): void
+    {
+        $this->client->sort(['createdAt' => 'oldest' === $sortDirection ? 'asc' : 'desc']);
```

Comment:
maybe it is only my private preferences but making it as a two separate steps will be more readable for me :)